lib/carbon/writer.py

```diff
@@ -57,12 +62,21 @@ def writeCachedDataPoints():
 
   cache = MetricCache()
   while cache:
-    (metric, datapoints) = cache.drain_metric()
+    if (shuttingDown == 1) or (MetricCache.size >= LOW_CACHE_SIZE):
+      mdpu = False
+    else:
+      mdpu = True
+    (metric, datapoints) = cache.drain_metric(mdpu)
     if metric is None:
       # end the loop
       break
-
-    dbFileExists = state.database.exists(metric)
+    if state.cacheTooFull and MetricCache.size < settings.CACHE_SIZE_LOW_WATERMARK:
```

Comment:
Looks like it's deprecated now.  cache.drain_meric() will call cache.pop() which has this check already - https://github.com/graphite-project/carbon/blob/9fcb6f2c2acb6361a7461880f2373b085e5c42a2/lib/carbon/cache.py#L201